packages/react-components/priority-overflow/src/overflowManager.ts

```diff
@@ -33,6 +36,60 @@ export function createOverflowManager(): OverflowManager {
     update();
   });
 
+  function updateGroupVisibility(groupId: string) {
+    const group = groups[groupId];
+    if (group.invisibleItemIds.size && group.visibleItemIds.size) {
+      groupVisibility[groupId] = 'overflow';
+    } else if (group.visibleItemIds.size === 0) {
+      groupVisibility[groupId] = 'hidden';
+    } else {
+      groupVisibility[groupId] = 'visible';
+    }
+  }
+
+  const getNextItem = (queueToDequeue: PriorityQueue<string>, queueToEnqueue: PriorityQueue<string>) => {
```

Comment:
this is a pure function - you can create a utils file/folder and store it there

Reply:
But then we'll need to pass `overflowItems` as a parameter cause this function needs it

Reply:
added this idea to the refactoring issue: #28236